### PR TITLE
trains prefer to stay in lane at all intersections

### LINKED ISSUE #1475: trains switch lanes too much
Part of this : #1223

~~If the asset designer disables `CanCrossLanes` then the tracks should go to their respective lanes.~~
[EDIT] Trains switch change too much by default both at intersection and middle lanes. I think by default trains should not switch lanes too much like this:
image
image

and not like this (created with AN mod):
image

User should be able to override this behavior and change track transitions.

### PR DESCRIPTION
**Pull request by kianzarrin**
fixes #1475
[TMPE.zip](https://ci.appveyor.com/api/projects/krzychu124/tmpe/artifacts/TMPE.zip?branch=1475-track-stayinlane)

- I check if the lane is a track only lane. that is train/metro/monorail/tram  but mixed tram/car does not count. 
- I also check if both source/target segments have equal number of tracks.
- if both of the above conditions are met then : Routing manager only creates one lane transition which is between the two lanes with equal `similar Lane Index`.
- user can override the above behavior using lane connections

PS: Created `TrackUtils` class even though it has only one function. I might add more stuff to it in future if I deal with Continues junction barrier stuff (#1223) or if I bring Hide-Unconnectred-Tracks to TMPE. (maybe I should call it DirectConnectUtils)
PS2:  I used `$"{varaible}"` to log stuff because it was more reliable. I hope you don't mind: https://github.com/CitiesSkylinesMods/TMPE/discussions/1484

EDIT: for bidirectional/station tracks I fall back to original code.

### CODE REVIEW

**Comment by kianzarrin** [comment]
seems to work: 
image

**Comment by kianzarrin** [comment]
Test lane connection can be used to change lane connections ✔️ 
image

File: TLM/TLM/Util/TrackUtils.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        internal const NetInfo.LaneType LANE_TYPES =
+            NetInfo.LaneType.Vehicle | NetInfo.LaneType.TransportVehicle;
+
+        internal const VehicleInfo.VehicleType VEHICLE_TYPES =
+            VehicleInfo.VehicleType.Metro |
+            VehicleInfo.VehicleType.Train |
+            VehicleInfo.VehicleType.Tram |
+            VehicleInfo.VehicleType.Monorail;
+
+        internal static bool IsTrackOnly(this NetInfo.Lane laneInfo) {
```
[the remark this answers is not in the document]
Comment (kianzarrin) [reply TLM/TLM/Util/TrackUtils.cs:14]: its a class. that is why it can be null.

**Review by kvakvs — APPROVED** [review]
Code looking fine

**Comment by kianzarrin** [comment]
I will ignore bi-directional lanes in this PR. I created  #1486 to cover bi-directional problems for all of routing manager (note that the code already had problem with bidirectional lanes even before this PR).

**Comment by kianzarrin** [comment]
Its working as expected now: all lane transitions are created for bi-directional tracks
image

**Comment by aubergine10** [comment]
will this PR benefit from some of the extensions you added in #1487

**Comment by kianzarrin** [comment]
@aubergine10 
> will this PR benefit from some of the extensions you added in #1487

No. All I need here is to ignore train/bidirectional lanes which I can detect using `IsBidirectional()`. I have deleted that in the next PR because its  just a workaround for work that is outside of scope of this PR (https://github.com/CitiesSkylinesMods/TMPE/pull/1483#issuecomment-1075309041).

In any case if you think the code is good you can approve this PR.

**Review by krzychu124 — APPROVED** [review]

**Pull request merged**